ping-payload-extensions/ping-custom-payload-extensions.md

```diff
@@ -0,0 +1,62 @@
+# Ping Custom Payload Extensions
+
+## Motivation
+
+Ping Payload Extensions. Messages on Portal are primarily made of ping/pong responses. This framework allows Portal clients to implement `non standard extensions` which don't require a breaking change to deploy to the network. A more flexible way to extend the Protocol without bloating the [Portal-Wire-Protocol](../portal-wire-protocol.md) or requiring every client to agree add a feature a subset of clients want.
+
+# Type's
+
+There are 65536 unique type ids.
+
+Types 0-10_000 and 65436-65535 are reserved for for future upgrades.
+
+The rest are first come first serve, but they should still be defined in this repo to avoid overlaps.
+
+
+## Requirements
+
+All payloads used in the Ping `custom_payload` MUST follow the `Ping Custom Payload Extensions` format.
+
+## Custom Payload Extensions Format
+
+- **type**: numeric identifier which tells clients how the `payload` field should be decoded.
+- **payload**: the SSZ encoded extension payload
+
+
+```python
+CustomPayloadExtensionsFormat = Container(
+  type: u16,
+  payload: ByteList[max_length=1100]
+)
+```
+
+## Ping vs Pong
+The relationship between Ping and Pong message will be determined by the requirements of the type.
+
+Currently type 0, 1, and 2 are symmetrical, having the same payload for both request and response. This symmetry is not required. Extensions may define separate payloads for Ping and Pong within the same type.
+
+
+### Error responses
+If the ping receiver can't handle the ping for any reason the pong should return an error payload
+
+[Type 65535: The definition of error responses can be found here](extensions/type-65535.md)
+
+## Standard extensions
+
+A standard extension is an extension which all nodes on the network can assume all other clients support, without requiring a `Type 0: Client Info, Radius, and Capabilities Payload` to discover what extensions the client supports. 
+
+Changing standard extensions is considered a breaking change.
+
+List of some standard extensions
+- [Type 0: Client Info, Radius, and Capabilities Payload](extensions/type-0.md): useful for finding Client Info, Radius, and ping extensions a client supports
+
```

Comment:
Should we not add type `65535` also as a standard extension? All nodes MUST be able to return errors...